public/app/features/alerting/unified/hooks/usePrometheusConsistencyCheck.ts

```diff
@@ -43,6 +52,172 @@ function useMatchingPromRuleExists() {
   return { matchingPromRuleExists };
 }
 
+const PREFER_CACHE_VALUE = true;
+
+export function useRuleGroupIsInSync() {
+  const [discoverDsFeatures] = useLazyDiscoverDsFeaturesQuery();
+  const [fetchPrometheusRuleGroups] = useLazyPrometheusRuleNamespacesQuery();
+  const [fetchRuleGroup] = useLazyGetRuleGroupForNamespaceQuery();
+
+  const isGroupInSync = useCallback(
```

Comment:
Shouldn't we use `useAsync` here so we can track progress for this check somehow?